src/Carousel.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect, useRef } from 'react';
+import './Carousel.css';
+
+function Carousel() {
```

Comment:
All function better use like this: 

const Carousel = () => {

}

Because you declare variable and the memory area is immediately allocated for the function, and if you write like this, then JS under the hood does it itself.